packages/react-scripts/config/paths.js

```diff
@@ -40,6 +41,37 @@ var nodePaths = (process.env.NODE_PATH || '')
   .filter(folder => !path.isAbsolute(folder))
   .map(resolveApp);
 
+var envPublicUrl = process.env.PUBLIC_URL;
+
+function ensureSlash(path, needsSlash) {
+  var hasSlash = path.endsWith('/');
+  if (hasSlash && !needsSlash) {
+    return path.substr(path, path.length - 1);
```

Comment:
Shouldn't this have been `path.length`?

Reply:
Agreed that this is wrong, though I think that it should be `0`, not `path.length`. This code works very accidentally, because `Number(path)` is `NaN`, which is converted to `+0` by `ToInteger` in the definition of `String.prototype.substr`:
  - [ECMAScript docs for `substr`](https://www.ecma-international.org/ecma-262/6.0/)
  - [ECMAScript docs for `ToInteger`](https://www.ecma-international.org/ecma-262/6.0/#sec-tointeger)

Reply:
Fixed in #4424.